client/src/lib.rs

```diff
@@ -503,23 +514,36 @@ pub enum ClientError {
     IOError(#[from] std::io::Error),
 }
 
+#[cfg(feature = "async")]
+pub trait ThreadSafeSigner: Signer + Send + Sync + 'static {
+    fn as_signer(&self) -> &dyn Signer;
+}
```

Comment:
Only commenting here, but this applies to all `#[cfg(feature = "async")]` checks in this file. It would be better for them to just exist in `nonblocking.rs` instead of `lib.rs`, since they are only being used there.